### PR TITLE
[master] Add Query Parameter Annotation Mapping for Http Service

### PR DESCRIPTION
**Pull request by lnash94**
## Purpose
Fix: https://github.com/ballerina-platform/ballerina-library/issues/7006
Partial fix for the main issue: https://github.com/ballerina-platform/ballerina-library/issues/6747

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] Updated the spec https://github.com/ballerina-platform/ballerina-library/issues/6987
- [ ] Checked native-image compatibility
- [x] Checked the impact on OpenAPI generation - Fixed via https://github.com/ballerina-platform/openapi-tools/pull/1771

### CODE REVIEW

File: ballerina-tests/http-dispatching-tests/tests/service_dispatching_query_param_binding_test.bal [hunk 1]
```diff
@@ -167,8 +167,29 @@ service /queryparamservice on QueryBindingEP {
...
+    }
+
+    resource function get queryAnnotation/negative/q2(@http:Query{ name: ()} string lastName) returns  string {
+        return string `Hello, ${lastName}`;
+    }
+
+    resource function get queryAnnotation/mapQueries(@http:Query{ name: "rMq"} Mq mq) returns  string {
+        return string `Hello, ${mq.name} ${mq.age}`;
+    }
 }
```
Comment (NipunaRanasinghe) [inline ballerina-tests/http-dispatching-tests/tests/service_dispatching_query_param_binding_test.bal:186]: please fix code formatting

**Comment by NipunaRanasinghe** [comment]
@lnash94 do we need any updates to the tool spec along with these changes?

File: native/src/main/java/io/ballerina/stdlib/http/api/service/signature/ParamHandler.java [hunk 2]
```diff
@@ -272,11 +274,28 @@ private void createHeaderParam(String paramName, BMap annotations) {
...
+        QueryParam queryParam = new QueryParam(originalType, paramName, index, parameter.isDefault,
+                constraintValidation);
         this.queryParams.add(queryParam);
     }
 
+    private static String getQueryParamName(String paramName, BMap annotations) {
+        BMap mapValue = annotations.getMapValue(StringUtils.fromString(QUERY_ANNOTATION));
+        Object queryName = mapValue.get(HttpConstants.ANN_FIELD_NAME);
+        if (queryName instanceof BString query) {
+            paramName = query.getValue();
```
Comment (MohamedSabthar) [inline native/src/main/java/io/ballerina/stdlib/http/api/service/signature/ParamHandler.java:294]:
```suggestion
            return query.getValue();
```

**Comment by lnash94** [comment]
> @lnash94 do we need any updates to the tool spec along with these changes?

These changes need to be addressed via HTTP-side spec and annotation integrating changes, which we will address via the tool specs.

File: ballerina-tests/http-dispatching-tests/tests/service_dispatching_query_param_binding_test.bal [hunk 3]
```diff
@@ -322,11 +343,11 @@ function testMapJsonOfDefaultableQueryBinding() returns error? {
...
-        objects: [
-            {name: "test1", value: "json1"},
-            {name: "test2", value: "json2"}
-        ]
-    });
+                                                       objects: [
+                                                           {name: "test1", value: "json1"},
+                                                           {name: "test2", value: "json2"}
+                                                       ]
+                                                   });
```
Comment (NipunaRanasinghe) [inline ballerina-tests/http-dispatching-tests/tests/service_dispatching_query_param_binding_test.bal:350]: Is this intentional? for me, the previous formatting looks better.

Comment (lnash94) [reply ballerina-tests/http-dispatching-tests/tests/service_dispatching_query_param_binding_test.bal:350]: This was formatted using the `bal format` command and was not intentionally changed; it was kept as it was returned by the command. Do we need to revert it to the previous format?

**Review by TharmiganK — DISMISSED** [review]
LGTM

File: ballerina-tests/http-dispatching-tests/tests/service_dispatching_query_param_binding_test.bal (hunk 3 above)
Comment (lnash94) [reply ballerina-tests/http-dispatching-tests/tests/service_dispatching_query_param_binding_test.bal]: Address via [e3855c5](https://github.com/ballerina-platform/module-ballerina-http/pull/2152/commits/e3855c53a29907ccda8f2ae93758e028c99b6922)

**Pull request merged**